src/install/installer.ts

```diff
@@ -52,8 +52,13 @@ async function validateCache(packagePath: string, browsersPath: string, linksDir
     let linkTarget = '';
     try {
       linkTarget = (await fsReadFileAsync(linkPath)).toString();
-      const browsers = JSON.parse((await fsReadFileAsync(path.join(linkTarget, 'browsers.json'))).toString())['browsers'];
+      const browsers = JSON.parse((await fsReadFileAsync(path.join(linkTarget, 'browsers.json'))).toString())['browsers'] as browserPaths.BrowserDescriptor[];
+      // Older versions do not have "download" field. We assume they need all browsers
+      // from the list. For newer ones, only the browsers with "download" are counted.
```

Comment:
I don't really like this backward-compat dance. How about baking the knowledge of what browsers to download using package name inside `installBrowsersWithProgressBar`?

I think this way the logic might be simplified quite a lot.